File: 1399. Count Largest Group/solution.cpp
class Solution {
public:
  int countLargestGroup(int n) {
    unordered_map<int, int> dictionary;

    for (int index = 1; index < n + 1; index++){
      int total = 0;
      string stringIndex = to_string(index);
      for (int charIndex = 0; charIndex < stringIndex.size(); charIndex++){
        total += (int) stringIndex[charIndex] - 48;
      }
      dictionary[total]++;
    }

    int totalCount = 0;
    int highestCount = 0;
    for (auto num : dictionary){
      if (num.second > highestCount){
        highestCount = num.second;
        totalCount = 1;
      }
      else if (num.second == highestCount){
        totalCount += 1;
      }
    }

    return totalCount;
  }
};

File: 1399. Count Largest Group/solution.py
class Solution(object):
  def countLargestGroup(self, n):
    """
    :type n: int
    :rtype: int
    """
    dictionary = {}

    for index in range(1, n + 1):
      total = 0
      for value in str(index):
        total += int(value)
        
      if total not in dictionary:
        dictionary[total] = 1
      else:
        dictionary[total] += 1
    
    totalCount = 0
    highestCount = 0
    for key in dictionary:
      if dictionary[key] > highestCount:
        highestCount = dictionary[key]
        totalCount = 1
      elif dictionary[key] == highestCount:
        totalCount += 1

    return totalCount